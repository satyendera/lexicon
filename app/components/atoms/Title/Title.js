// @flow
import React from 'react';
import styles from './Title.style';
import withStyles from '../../../lib/withStyles';



const Title = ({ children, className, inheritedStyles, ...others }) => (
  <h1 className={className} {...others}>
    {children}
  </h1>
);

Title.defaultProps = {
  inheritedStyles: '',
};

export default withStyles(Title, styles);
export { Title as TitleVanilla };
