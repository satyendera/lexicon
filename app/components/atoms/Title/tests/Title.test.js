import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { TitleVanilla as Title } from '../Title';

const props = {
  children: 'Sample',
  className: '',
};
describe('Anchor Component', () => {
  let title;

  test('should render correctly and match snapshot', () => {
    title = shallow(<Title {...props} />);
    expect(title).toMatchSnapshot();
  });

  test('should render with styles correctly and match snapshot', () => {
    title = renderer.create(<Title {...props} />).toJSON();
    expect(title).toMatchSnapshot();
  });
});
