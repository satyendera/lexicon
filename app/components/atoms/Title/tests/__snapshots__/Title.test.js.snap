// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Component should render correctly and match snapshot 1`] = `
<h1
  className=""
>
  Sample
</h1>
`;

exports[`Anchor Component should render with styles correctly and match snapshot 1`] = `
<h1
  className=""
>
  Sample
</h1>
`;
